src/utils/identifyOnVectorLayer.js

```diff
@@ -169,14 +150,15 @@ export function identifyKMLFeatureAt(kmlLayer, coordinate, projection, resolutio
         const convertedKml = kmlToGeoJSON(parseKml)
         return identifyInGeoJson(convertedKml, coordinate, projection, resolution).map(
             (feature) => {
-                // TODO use a new generic feature for identify (same for external KML and GPX) and not the editable
-                return EditableFeature.newFeature({
-                    id: feature.id,
-                    coordinates: reprojectCoordinates(feature.geometry.coordinates, projection),
-                    title: feature.properties.name,
-                    description: feature.properties.description,
-                    featureType: feature.properties.type.toUpperCase(),
-                })
+                return new LayerFeature(
+                    kmlLayer,
+                    feature.id,
+                    feature.properties.name,
+                    generateHtmlPopup(kmlLayer.name, feature.properties.description),
```

Comment:
This is way too dangerous and subject to XSS attack we cannot trust the feature description content that comes from users including geoadmin KML, user can enter their own html code in a kml description. We should rethink this component and use an HTML sanitizer like https://github.com/cure53/DOMPurify

Reply:
I've changed how we render the HTML for features, and if we receive a string value from the feature I sanitize it before injecting it (with this DOMPurify lib you found).